### PR TITLE
[ADD] Room model

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: resource_calendar/README.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+This module contains the dependencies, configuration and data to setup a
+database and test Client configuration.
+
+Contributors
+------------
+* Luis Garcia (luis.garcia@savoirfairelinux.com)
+
+More information
+----------------
+* Module developed and tested with Odoo version 8.0
```
Comment (rimbendhaou) [inline resource_calendar/README.rst]: @eilst version 11.0

File: resource_calendar/i18n/fr.po [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+#	* client
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 8.0\n"
```
Comment (rimbendhaou) [inline resource_calendar/i18n/fr.po]: @eilst version 11.0

File: resource_calendar/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (rimbendhaou) [inline resource_calendar/models/__init__.py]: @eilst license lgpl

File: resource_calendar/models/instrument.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst license lgpl

File: resource_calendar/models/instrument.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+from odoo import _, api, fields, models
+
+class Instrument(models.Model):
+    _inherit = ['resource.resource']
+    _name = 'resource_calendar.instrument'
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst resource.calendar.instrument

File: resource_calendar/models/resource.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (rimbendhaou) [inline resource_calendar/models/resource.py]: @eilst licenses

File: resource_calendar/models/resource.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
```
Comment (rimbendhaou) [inline resource_calendar/models/resource.py]: @eilst no need

File: resource_calendar/models/room.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst license

File: resource_calendar/models/room.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+from odoo import _, api, fields, models
+
+class Room(models.Model):
+    _inherit = ['resource.resource']
+    _name = 'resource_calendar.room'
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst resource.calendar.room

File: resource_calendar/__manifest__.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+{
+    'name': 'Resource Calendar',
+    'version': '1.0',
+    'author': 'Savoir-faire Linux',
+    'maintainer': 'Savoir-faire Linux',
+    'website': 'http://www.savoirfairelinux.com',
+    'license': 'LGPL-3',
+    'category': 'Others',
```
Comment (rimbendhaou) [inline resource_calendar/__manifest__.py]: @eilst Project

File: resource_calendar/models/instrument.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+    name = fields.Char(
+        string='Name',
+    )
+    room_id = fields.Many2one(
+        string='Room',
+        comodel_name='resource_calendar.room',
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst resource.calendar.room

File: resource_calendar/models/instrument.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+        inverse_name='name'
+    )
+    resource_id = fields.One2many(
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst if One2many field it should be resource_ids

File: resource_calendar/models/instrument.py [hunk 13]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+
+class Instrument(models.Model):
+    _inherit = ['resource.resource']
+    _name = 'resource_calendar.instrument'
+
+    name = fields.Char(
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst name field already exist

File: resource_calendar/models/instrument.py [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+        string='Name',
+    )
+    room_id = fields.Many2one(
+        string='Room',
+        comodel_name='resource_calendar.room',
+        inverse_name='name'
+    )
+    resource_id = fields.One2many(
+        string='Resource Id',
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst Resources

File: resource_calendar/models/room.py [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+
+    id_room = fields.Char(
+        string='Room ID',
+    )
+    name = fields.Char(
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst field name already exist

File: resource_calendar/models/room.py [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+    )
+    capacity = fields.Integer(
+        string='Capacity',
+    )
+    description = fields.Text(
+        string='Dsescription',
+    )    
+    room_type = fields.Many2one(
+        string='Room Type',
+        comodel_name='resource_calendar.room_type',
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: resource.calendar.room.type

File: resource_calendar/models/room.py [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+    name = fields.Char(
+        string='Name'
+    )
+    capacity = fields.Integer(
+        string='Capacity',
+    )
+    description = fields.Text(
+        string='Dsescription',
+    )    
+    room_type = fields.Many2one(
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst room_type_id a Many_2one field should end with _id

File: resource_calendar/models/room.py [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+        string='Room Type',
+        comodel_name='resource_calendar.room_type',
+        ondelete='set null',
+    )
+    is_bookable = fields.Boolean(
+        string='Is Bookable',
+    )
+    sector = fields.Many2one(
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst sector_id (Many2one field)

File: resource_calendar/models/room.py [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+        string='Sector',
+        comodel_name='resource_calendar.sector',
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: resource.calendar.sector

File: resource_calendar/models/room.py [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+        ondelete='set null',
+    )
+    instruments_ids = fields.Many2one(
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst instrument_id (Many2one field)

File: resource_calendar/models/room.py [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+        'resource_calendar.instrument',
+        string='Instrument'
+    )
+
+    resource_id = fields.One2many(
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst resource_ids

File: resource_calendar/models/instrument.py (hunk 12 above)
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst no need

File: resource_calendar/models/instrument.py [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+from odoo import _, api, fields, models
+
+class Instrument(models.Model):
+    _inherit = ['resource.resource']
+    _name = 'resource_calendar.instrument'
+
+    name = fields.Char(
+        string='Name',
+    )
+    room_id = fields.Many2one(
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst you already add room_id field in the resource.resource

File: resource_calendar/models/resource.py [hunk 23]
```diff
@@ -0,0 +1,15 @@
...
+class Resource(models.Model):
+    _inherit = 'resource.resource'
+
+    resource_type = fields.Selection([
+        ('user', 'Human'),
+        ('material', 'Material'),
+        ('room', 'Room')], string='Resource Type',
+        default='user', required=True)
+
+    room_id = fields.Many2one('resource_calendar.room', string='Room')
```
Comment (rimbendhaou) [inline resource_calendar/models/resource.py]: @eilst Why you need this, you also added room_id in the resource.resource.instrument

File: resource_calendar/models/room.py [hunk 24]
```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+
+class Room(models.Model):
+    _inherit = ['resource.resource']
+    _name = 'resource_calendar.room'
+
+    id_room = fields.Char(
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst room_code

File: resource_calendar/models/room.py (hunk 20 above)
Comment (rimbendhaou) [reply resource_calendar/models/room.py]: @eilst are you sure we have only one intrument in each room?

File: resource_calendar/models/room.py [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+    )
+    instruments_ids = fields.Many2one(
+        'resource_calendar.instrument',
+        string='Instrument'
+    )
+
+    resource_id = fields.One2many(
+        string='Resource Id',
+        comodel_name='resource.resource',
+        inverse_name='name',
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst room_id

File: resource_calendar/models/room.py [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+    )
+    # TODO: Pricind field model ? field
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst ??

File: resource_calendar/models/room_type.py [hunk 27]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (rimbendhaou) [inline resource_calendar/models/room_type.py]: @eilst license

File: resource_calendar/models/room.py [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+
+    @api.model
+    def create(self, values):
+        """
+            Create a new record for a model Room
+            @param values: provides a data for new record
+            @return: returns a id of new record
+        """
+        values['resource_type'] = 'room'
+        result = super().create(values)
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst super()?

File: resource_calendar/models/room.py [hunk 29]
```diff
@@ -0,0 +1,56 @@
...
+
+    resource_id = fields.One2many(
+        string='Resource Id',
+        comodel_name='resource.resource',
+        inverse_name='name',
+    )
+    # TODO: Pricind field model ? field
+
+    @api.model
+    def create(self, values):
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst it is correct but better to use vals in place of values

File: resource_calendar/models/room_type.py [hunk 30]
```diff
@@ -0,0 +1,11 @@
...
+from odoo import _, api, fields, models
```
Comment (rimbendhaou) [inline resource_calendar/models/room_type.py]: @eilst _ is not used

File: resource_calendar/models/sector.py [hunk 31]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
```
Comment (rimbendhaou) [inline resource_calendar/models/sector.py]: @eilst _ imported but not used

File: resource_calendar/views/instrument.xml [hunk 32]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rimbendhaou) [inline resource_calendar/views/instrument.xml]: @eilst empty?

File: resource_calendar/views/webclient_templates.xml [hunk 33]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rimbendhaou) [inline resource_calendar/views/webclient_templates.xml]: @eilst no need, we use that only for a new project for clients

File: resource_calendar/models/sector.py [hunk 34]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (rimbendhaou) [inline resource_calendar/models/sector.py]: @eilst you should add model access in ir.model.access.csv

File: resource_calendar/models/sector.py [hunk 35]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (rimbendhaou) [inline resource_calendar/models/sector.py]: @eilst license

File: resource_calendar/models/room.py [hunk 36]
```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
```
Comment (rimbendhaou) [inline resource_calendar/models/room.py]: @eilst _ imported but not used

File: resource_calendar/models/resource.py [hunk 37]
```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
```
Comment (rimbendhaou) [inline resource_calendar/models/resource.py]: @eilst _ imported but not used

File: resource_calendar/models/instrument.py [hunk 38]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Savoir-faire Linux
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst _ not used, did you used api?

File: resource_calendar/data/res_partner_timezone_admin.xml [hunk 39]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rimbendhaou) [inline resource_calendar/data/res_partner_timezone_admin.xml]: @eilst no need, we need it only for new project

File: resource_calendar/__manifest__.py (hunk 10 above)
Comment (eilst) [reply resource_calendar/__manifest__.py]: Project? I don't get this one

File: resource_calendar/README.rst [hunk 40]
```diff
@@ -0,0 +1,14 @@
+Client
+======
+This module contains the dependencies, configuration and data to setup a
```
Comment (rimbendhaou) [inline resource_calendar/README.rst]: @eilst this is no the description of the module

File: resource_calendar/README.rst [hunk 41]
```diff
@@ -0,0 +1,14 @@
+Client
```
Comment (rimbendhaou) [inline resource_calendar/README.rst]: @eilst you should put the name of the module

File: resource_calendar/__manifest__.py [hunk 42]
```diff
@@ -0,0 +1,31 @@
...
+    'summary': 'resource calendar',
+    'depends': [
+        'calendar_resource',
+        "resource",
```
Comment (rimbendhaou) [inline resource_calendar/__manifest__.py]: @eilst better to put 'resource', 
                               'calendar',

File: resource_calendar/i18n/fr.po [hunk 43]
```diff
@@ -0,0 +1,27 @@
+# Translation of Odoo Server.
```
Comment (rimbendhaou) [inline resource_calendar/i18n/fr.po]: @eilst you have to update the translation

File: resource_calendar/i18n/fr.po [hunk 44]
```diff
@@ -0,0 +1,27 @@
...
+# This file contains the translation of the following modules:
+#	* client
```
Comment (rimbendhaou) [inline resource_calendar/i18n/fr.po]: @eilst the module name is not client

File: resource_calendar/models/instrument.py [hunk 45]
```diff
@@ -0,0 +1,31 @@
...
+from odoo import fields, models
+
+
+class Instrument(models.Model):
+    _inherit = ['resource.resource']
+    _name = 'resource.calendar.instrument'
+
+    room_id = fields.Many2one(
+        'resource.calendar.room',
+        string='Room',)
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst new line after ,

File: resource_calendar/models/instrument.py [hunk 46]
```diff
@@ -0,0 +1,31 @@
...
+        string='Resources',
+        comodel_name='resource.resource',
+        inverse_name='name',
+    )
+    sku = fields.Char(
+        string='Sku',
+    )
+    type_id = fields.Many2one(
+        string=u'type_id',
+        comodel_name='model.name',
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst comodel_name='model.name'?

File: resource_calendar/models/instrument.py [hunk 47]
```diff
@@ -0,0 +1,31 @@
...
+    room_id = fields.Many2one(
+        'resource.calendar.room',
+        string='Room',)
+    resource_ids = fields.One2many(
+        string='Resources',
+        comodel_name='resource.resource',
+        inverse_name='name',
+    )
+    sku = fields.Char(
+        string='Sku',
```
Comment (rimbendhaou) [inline resource_calendar/models/instrument.py]: @eilst capitalize (SKU)

File: project_event/models/project_task.py [hunk 48]
```diff
@@ -68,13 +68,55 @@ class Task(models.Model):
...
+        ondelete='set null',
+    )
+    resource_type = fields.Selection([
+        ('user', 'Human'),
+        ('equipment', 'Equipment'),
+        ('room', 'Room')], string='Resource Type',
+        default='room', required=True)
+
+    @api.onchange('room_id')
+    def _ensure_one_resource_room(self):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: def onchange_room_id

File: project_event/models/project_task.py [hunk 49]
```diff
@@ -68,13 +68,55 @@ class Task(models.Model):
...
+        if self.room_id:
+            self.equipment_id = None
+        elif self.equipment_id:
+            self.room_id = None
+
+    @api.onchange('equipment_id')
+    def _ensure_one_resource_equipment(self):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @eilst def onchange_equipment_id(self):

File: project_event/models/project_task.py [hunk 50]
```diff
@@ -68,13 +68,55 @@ class Task(models.Model):
...
+        string='Equipment',
+        comodel_name='resource.calendar.instrument',
+        ondelete='set null',
+    )
+    resource_type = fields.Selection([
+        ('user', 'Human'),
+        ('equipment', 'Equipment'),
+        ('room', 'Room')], string='Resource Type',
+        default='room', required=True)
+
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:93]: @eilst I think better to put def onchange_resource_type (self):
self.room_id = Flase
self.equipment_id = False

File: resource_calendar/models/sector.py [hunk 51]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2018 Savoir-faire Linux
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import fields, models
+
+
+class Sector(models.Model):
+    _name = 'resource.calendar.sector'
```
Comment (rimbendhaou) [inline resource_calendar/models/sector.py]: @eilst sector.sector because it is not only related to resource calendar

**Review by rimbendhaou — COMMENTED** [review]
Please change the name of the resource_calendar to project_resource_calendar and add the translation when you finish

File: project_resource_calendar/security/ir.model.access.csv [hunk 52]
```diff
@@ -0,0 +1,6 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_room,access_room,model_resource_calendar_room,base.group_user,1,1,1,1
+access_instrument,access_instrument,model_resource_calendar_instrument,base.group_user,1,1,1,1
+access_sector,access_sector,model_resource_calendar_sector,base.group_user,1,1,1,1
```
Comment (rimbendhaou) [inline project_resource_calendar/security/ir.model.access.csv]: @eilst model_resource_calendar_sector? it is model_sector_sector

**Review by rimbendhaou — APPROVED** [review]
LGTM

**Pull request merged**